HLTrigger/Timer/src/FastTimerService.cc

```diff
@@ -250,6 +217,42 @@ void FastTimerService::preStreamBeginRun(edm::StreamContext const & sc)
   unsigned int sid = sc.streamID().value();
   auto & stream = m_stream[sid];
 
+  edm::service::TriggerNamesService & tns = * edm::Service<edm::service::TriggerNamesService>();
+
+  uint32_t size_p = tns.getTrigPaths().size();
+  uint32_t size_e = tns.getEndPaths().size();
+  uint32_t size = size_p + size_e;
+
+  // resize the path maps
+  if (stream.paths.size() <= pid)
+    stream.paths.resize(pid+1);
+  for (uint32_t i = 0; i < size_p; ++i) {
+    std::string const & label = tns.getTrigPath(i);
+    stream.paths[pid][label].index = i;
+  }
+  for (uint32_t i = 0; i < size_e; ++i) {
+    std::string const & label = tns.getEndPath(i);
+    stream.paths[pid][label].index = size_p + i;
+  }
+
+  // resize the stream buffers to account the number of subprocesses
+  if (stream.timing_perprocess.size() <= pid)
+    stream.timing_perprocess.resize(pid+1);
+  stream.timing_perprocess[pid].paths_interpaths.assign(size + 1, 0.);
+  // resize the stream plots to account the number of subprocesses
+  if (stream.dqm_perprocess.size() <= pid)
+    stream.dqm_perprocess.resize(pid+1);
+  if (stream.dqm_perprocess_byluminosity.size() <= pid)
+    stream.dqm_perprocess_byluminosity.resize(pid+1);
+  if (stream.dqm_paths.size() <= pid)
+    stream.dqm_paths.resize(pid+1);
+
+  // associate to each path all the modules it contains
+  for (uint32_t i = 0; i < tns.getTrigPaths().size(); ++i)
+    fillPathMapForStream( stream,pid, tns.getTrigPath(i), tns.getTrigPathModules(i) );
+  for (uint32_t i = 0; i < tns.getEndPaths().size(); ++i)
+    fillPathMapForStream( stream, pid, tns.getEndPath(i), tns.getEndPathModules(i) );
+
```

Comment:
It appeared to be resetting stream related quantities. However, a begin global run would run before a stream end run (for the previous run) so it is not good to communicate with any stream related quantities. It would also be possible for multiple begin global runs to be called with different streams working on different runs, which also means global begin transitions should not look at stream related quantities.